includes/html/api_functions.inc.php

```diff
@@ -2586,6 +2586,16 @@ function del_service_from_host(Illuminate\Http\Request $request)
     return api_error(500, 'Failed to delete the service');
 }
 
+function search_by_mac(Illuminate\Http\Request $request)
+{
+    $search = $request->route('search');
+    $ports = dbFetchRows('select ports.* from ports_fdb join ports on ports.port_id=ports_fdb.port_id join devices on devices.device_id=ports.device_id where ports_fdb.port_id in (select port_id from ports_fdb where mac_address=?) group by ports_fdb.port_id having count(ports_fdb.port_id)>0 order by count(ports_fdb.port_id) limit 1', $search);
```

Comment:
Would you mind writing this in eloquent syntax instead? It's more future proof. 
Not sure why you need to mix in device table either